oop1-construct.js

```diff
@@ -1,17 +1,115 @@
+        console.log('The vital parts of a class include a constructor function, its properties, and methods to influence its properties and more.');
+    }
+}
 
+const newWhyClass = new WhyClass('Classes exist for encapsulation of values, abstraction of their attributes, inheritance from other classes, and polymorphism to represent different values');
+newWhyClass.explain();
+newWhyClass.pieces();
 
 
 /*2. You are exploring the rainforests of the Amazon.  You have observed many different types of wildlife on your excursion.  To jog your memory, some different types of wildlife that you observed are found in the following URL : https://www.ietravel.com/blog/amazon-rainforest-animals-beginners-guide-21-species.  Feel free to use other resources online if you wish.
 
 Create a class that can categorize at least 6 of the creatures that you recall observing the Amazon.  Each creature should have 3 unique characteristics, and 2 unique actions that it can do. */
 
-
 //your code here...
+class AmazonAnimal {
+    constructor(name, bioClass, length){
+        this.name = name;
+        this.bioClass = bioClass;
+        this.length = length;
+    }
+}
+
+class Capybara extends AmazonAnimal {
+    eatGrass(){
+        console.log('Ate almost 8 pounds of grass today!');
+    }
+
+    congregate(){
+        console.log('Joined with almost 100 other capybaras!')
+    }
+}
+
+const capy = new Capybara('Capybara', 'Mammal', '4 feet');
 
 
+class Marmoset extends AmazonAnimal {
+    scavenge(){
+        console.log('Found some delicious tree sap, insects, and fruit!');
+    }
+
+    hide(){
+        console.log('Ducked out of sight from nosy humans.');
+    }
+}
+
+const marmo = new Marmoset('Pygmy Marmoset', 'Mammal', '5 inches');
+
+
+class Macaw extends AmazonAnimal {
+    soar(){
+        console.log('Flying far above the canopies!');
+    }
+
+    eatNut(){
+        console.log('Broke a coconut with my giant curved beak!');
+    }
+}
+
+const macaw = new Macaw('Hyacinth Macaw', 'Bird', '3 feet');
+
+
+class Toucan extends AmazonAnimal {
+    socialize(){
+        console.log('Flew with a few other toucans at sunset.');
+    }
+
+    findFood(){
+        console.log('Found all sorts of tasty treats, ranging from insects, fruit, reptiles, and even eggs!');
+    }
+}
+
+const toucan = new Toucan('Toco Toucan', 'Bird', '20 inches');
+
+
+class Anaconda extends AmazonAnimal {
+    consumePrey(){
+        console.log('Wrapped around a capybara and swallowed it whole.');
+    }
+
+    relax(){
+        console.log('Found a nice body of water to relax next to.');
+    }
+}
+
+const anaconda = new Anaconda('Green Anaconda', 'Reptile', '15 feet');
+
+
+class Turtle extends AmazonAnimal {
+    scavenge(){
+        console.log('Found some tasty leaves and fish!');
+    }
+
+    rest(){
+        console.log('Took a nap in the sun while tucked into my shell.');
+    }
+}
+
+const turtle = new Turtle('South American River Turtle', 'Reptile', '3 feet');
+
 
```

Comment:
Use the animals you construct to output to something to the console to show that your code works. (it does)